apps/main-app/src/components/dataTable/utils.js

```diff
@@ -79,6 +79,72 @@ export const getMultipleReferencesCuriesAndUrls = (references) => {
   return references.map((reference) => getSingleReferenceCurieAndUrl(reference));
 }
 
+const buildProvider = (annotation) => {
+  if(!annotation) return;
+  return {
+    dataProvider: annotation.dataProvider,
+    secondaryDataProvider: annotation.secondaryDataProvider
+  }
+}
+
+export const buildProviders = (annotations) => {
+  if(!annotations) return;
+  return annotations.map(annotation => {
+    return buildProvider(annotation);
+  });
+}
+
+export const buildProviderWithUrl = (annotation) => {
+
+  const { dataProvider, secondaryDataProvider } = buildProvider(annotation);
+
+  if(secondaryDataProvider){
+    return {
+      dataProvider: {
+          id: dataProvider.id,  
+          abbreviation: dataProvider.sourceOrganization?.abbreviation, 
+          url: buildSourceUrl(dataProvider)
+      }, 
+      secondaryDataProvider: { 
+          id: secondaryDataProvider.id,  
+          abbreviation: secondaryDataProvider.sourceOrganization?.abbreviation, 
+          url: buildSourceUrl(secondaryDataProvider)
+      }, 
+    }
+  } else {
+    return {
+      dataProvider: {
+          id: dataProvider.id,  
+          abbreviation: dataProvider.sourceOrganization?.abbreviation, 
+          url: buildSourceUrl(dataProvider)
+      }, 
+    }
+  }
+}
+
+export const buildProvidersWithUrl = (annotations) => {
+  if(!annotations) return;
+
+  return annotations.map((annotation) => {
+    return buildProviderWithUrl(annotation)
+  })
+}
+
+export const buildSourceUrl = (dataProvider) => {
+  if(!dataProvider) return;
+  let url;
+
+  //default to mod homepage if no crossReference is provided
+  if(!dataProvider.crossReference){
+    url = dataProvider.sourceOrganization?.homepageResourceDescriptorPage?.urlTemplate?.replace("[%s]", "");
```

Comment:
Does this take care of the home page? I'm not sure you would need to do a string replace here?

Reply:
It's needed for SGD. Their homepage URL from the API is `http://www.yeastgenome.org/[%s]`